Front-end/store/src/app/core/services/profile.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import {BehaviorSubject, Observable, Subject} from 'rxjs';
+
+import { map, share } from 'rxjs/operators';
+
+import { UserProfile } from '../models/user-profile';
+import { PROFILE_URL } from '../utils/values';
+import { UserProfileDto } from '../DTOs/user-profile-dto';
+import { UserProfileMapper } from '../mappers/user-profile.mapper';
+
+import { AuthorizationDataProvider } from './authorization-data.provider';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProfileService {
+
+  public isUserAuthorized$: Subject<boolean> = new Subject<boolean>();
+
+  constructor(
+    private readonly http: HttpClient,
+    private readonly userProfileMapper: UserProfileMapper,
+    private readonly authorizationDataProvider: AuthorizationDataProvider,
+  ) { }
+
+  public get userProfile(): UserProfile {
+    return JSON.parse(sessionStorage.getItem('profile') ?? '') as UserProfile;
```

Comment:
Really will be OK, if I get empty string, try parse it and convert to UserProfile?